packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -614,245 +639,105 @@ class ThemeData {
   String toString() => '$runtimeType(${ platform != defaultTargetPlatform ? "$platform " : ''}$brightness $primaryColor etc...)';
 }
 
-/// A lazily evaluated theme that provides the properties of the given
-/// [delegate] theme localized using the properties of the given
-/// [localTextGeometry].
-///
-/// The localization is done by merging of the [TextTheme] fields of the
-/// [delegate] into the [localTextGeometry] and caching the results.
-class _LocalizedThemeData implements ThemeData {
-  _LocalizedThemeData(this.delegate, this.localTextGeometry);
+class _IdentityThemeDataCacheKey {
+  _IdentityThemeDataCacheKey(this.baseTheme, this.localTextGeometry);
 
-  final ThemeData delegate;
+  final ThemeData baseTheme;
   final TextTheme localTextGeometry;
 
+  // TODO: use Jenkins hash function with identityHashCode
   @override
-  Color get accentColor => delegate.accentColor;
-
-  @override
-  Brightness get accentColorBrightness => delegate.accentColorBrightness;
-
-  @override
-  IconThemeData get accentIconTheme => delegate.accentIconTheme;
-
-  @override
-  Color get backgroundColor => delegate.backgroundColor;
-
-  @override
-  Brightness get brightness => delegate.brightness;
-
-  @override
-  Color get buttonColor => delegate.buttonColor;
-
-  @override
-  Color get canvasColor => delegate.canvasColor;
-
-  @override
-  Color get cardColor => delegate.cardColor;
-
-  @override
-  Color get dialogBackgroundColor => delegate.dialogBackgroundColor;
-
-  @override
-  Color get disabledColor => delegate.disabledColor;
-
-  @override
-  Color get dividerColor => delegate.dividerColor;
-
-  @override
-  Color get errorColor => delegate.errorColor;
-
-  @override
-  Color get highlightColor => delegate.highlightColor;
-
-  @override
-  Color get hintColor => delegate.hintColor;
-
-  @override
-  IconThemeData get iconTheme => delegate.iconTheme;
+  int get hashCode => _PrecomputedJenkins.finish(_PrecomputedJenkins.combine(identityHashCode(baseTheme), identityHashCode(localTextGeometry)));
 
   @override
-  Color get indicatorColor => delegate.indicatorColor;
-
-  @override
-  TargetPlatform get platform => delegate.platform;
-
-  @override
-  Color get primaryColor => delegate.primaryColor;
-
-  @override
-  Brightness get primaryColorBrightness => delegate.primaryColorBrightness;
-
-  @override
-  IconThemeData get primaryIconTheme => delegate.primaryIconTheme;
-
-  @override
-  Color get scaffoldBackgroundColor => delegate.scaffoldBackgroundColor;
-
-  @override
-  Color get secondaryHeaderColor => delegate.secondaryHeaderColor;
-
-  @override
-  Color get selectedRowColor => delegate.selectedRowColor;
-
-  @override
-  Color get splashColor => delegate.splashColor;
-
-  @override
-  Color get textSelectionColor => delegate.textSelectionColor;
+  bool operator ==(Object other) {
+    final _IdentityThemeDataCacheKey otherKey = other;
```

Comment:
assert that `other is _IdentityThemeDataCacheKey` and document that this object doesn't support being compared to other types (which normally would be a violation of `==` semantics)